lib/mail_room.rb

```diff
@@ -3,6 +3,13 @@
 require 'yaml'
 
 module MailRoom
+  def self.structured_logger
```

Comment:
I think the one major change I'd like is to get rid of this singleton and move this to configuration per-mailbox.

Reply:
This will take some figuring out, as there are places where you're using it before a mailbox exists. 🤔

Reply:
This would also allow setting different log paths for different mailboxes, or disabling some, which is probably one of my minimum feature requirements.

Reply:
I've done all this now, thanks for that feedback. I agree it's a much more flexible way to do this :)